server/graphql/clinician/src/query.rs

```diff
@@ -0,0 +1,262 @@
+            last_name,
+            initials,
+            registration_code,
+            category,
+            address1,
+            address2,
+            phone,
+            mobile,
+            email,
+            female,
+        } = self;
+
+        ClinicianFilter {
+            id: id.map(EqualFilter::from),
+            code: code.map(SimpleStringFilter::from),
+            first_name: first_name.map(SimpleStringFilter::from),
+            last_name: last_name.map(SimpleStringFilter::from),
+            initials: initials.map(SimpleStringFilter::from),
+            registration_code: registration_code.map(SimpleStringFilter::from),
+            category: category.map(SimpleStringFilter::from),
+            address1: address1.map(SimpleStringFilter::from),
+            address2: address2.map(SimpleStringFilter::from),
+            phone: phone.map(SimpleStringFilter::from),
+            mobile: mobile.map(SimpleStringFilter::from),
+            email: email.map(SimpleStringFilter::from),
+            female: female.or(Some(false)),
+        }
+    }
+}
+
+impl ClinicianConnector {
+    pub fn from_domain(names: ListResult<Clinician>) -> Self {
+        ClinicianConnector {
+            total_count: names.count,
+            nodes: names
+                .rows
+                .into_iter()
+                .map(ClinicianNode::from_domain)
+                .collect(),
+        }
+    }
+}
+
+impl ClinicianSortInput {
+    pub fn to_domain(self) -> ClinicianSort {
+        let key = match self.key {
+            ClinicianSortFieldInput::Code => ClinicianSortField::Code,
+            ClinicianSortFieldInput::FirstName => ClinicianSortField::FirstName,
+            ClinicianSortFieldInput::LastName => ClinicianSortField::LastName,
+            ClinicianSortFieldInput::Initials => ClinicianSortField::Initials,
+            ClinicianSortFieldInput::RegistrationCode => ClinicianSortField::RegistrationCode,
+            ClinicianSortFieldInput::Category => ClinicianSortField::Category,
+            ClinicianSortFieldInput::Address1 => ClinicianSortField::Address1,
+            ClinicianSortFieldInput::Address2 => ClinicianSortField::Address2,
+            ClinicianSortFieldInput::Phone => ClinicianSortField::Phone,
+            ClinicianSortFieldInput::Mobile => ClinicianSortField::Mobile,
+            ClinicianSortFieldInput::Email => ClinicianSortField::Email,
+            ClinicianSortFieldInput::Female => ClinicianSortField::Female,
+        };
+
+        ClinicianSort {
+            key,
+            desc: self.desc,
+        }
+    }
+}
+
+impl ClinicianNode {
+    pub fn from_domain(clinician: Clinician) -> Self {
+        ClinicianNode { clinician }
+    }
+
+    pub fn row(&self) -> &ClinicianRow {
+        &self.clinician.clinician_row
+    }
+}
+
+#[Object]
+impl ClinicianNode {
+    pub async fn id(&self) -> &str {
+        &self.row().id
+    }
+
+    pub async fn store(&self, ctx: &Context<'_>) -> Result<Option<StoreNode>> {
```

Comment:
What store does this refer to ?

Reply:
Yeah its not really useful (it was from the clinician row) I removed it since the `store_id` from the `store_join` is already know.